corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py

```diff
@@ -281,7 +281,10 @@ def update_open_and_deleted_ids(related):
     while next_ids:
         exclude = set(chain.from_iterable(seen_ix[id] for id in next_ids))
         with timing_context("get_related_indices({} cases, {} seen)".format(len(next_ids), len(exclude))):
-            related = get_related_indices(list(next_ids), exclude)
+            # filter out deleted indices
+            related = [
+                index for index in get_related_indices(list(next_ids), exclude) if index.referenced_id
```

Comment:
Do you think this filtering should go inside `get_related_indices`?  Looks like it's only used here.  Or even maybe in the SQL function

Reply:
Ha, just saw the next commit